Assets/LeapMotion/Scripts/DetectionUtilities/DetectorLogicGate.cs

```diff
@@ -0,0 +1,79 @@
+﻿using UnityEngine;
+using UnityEngine.Events;
+using System.Collections.Generic;
+
+namespace Leap.Unity {
+
+  public class DetectorLogicGate : Detector {
+    public List<Detector> Detectors;
+    public bool AddAllDetectorsOnAwake = true;
+    public LogicType GateType = LogicType.AndGate;
+    public bool Negate = false;
+
+    public void AddDetector(Detector detector){
+      if(!Detectors.Contains(detector)){
+        Detectors.Add(detector);
+      }
+    }
+
+    public void RemoveDetector(Detector detector){
+      detector.OnActivate.RemoveListener(CheckDetectors);
+      detector.OnDeactivate.RemoveListener(CheckDetectors);
+      Detectors.Remove(detector);
+    }
+
+    public void AddAllDetectors(){
+      Detector[] detectors = GetComponents<Detector>();
+      for(int g = 0; g < detectors.Length; g++){
+        if ( detectors[g].GetInstanceID() != this.GetInstanceID() && detectors[g].enabled) {
+          AddDetector(detectors[g]);
```

Comment:
We should check if the detector is already added, since a user might easily add a sibling detector component to the list.